### PR TITLE
Add workarounds for illegal `Task.getProject` access in AGP

### PR DESCRIPTION
**Pull request by h0tk3y**

### CODE REVIEW

**Comment by h0tk3y** [comment]
@bot-gradle test RFN

File: subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/TaskExecutionAccessCheckers.kt [hunk 1]
```diff
@@ -60,6 +60,13 @@ object TaskExecutionAccessCheckers {
     ) : AbstractTaskProjectAccessChecker(broadcaster) {
 
         override fun shouldReportExecutionTimeAccess(task: TaskInternal): Boolean =
-            !configurationTimeBarrier.isAtConfigurationTime
+            !configurationTimeBarrier.isAtConfigurationTime && !temporarilyIgnoreAsWorkaround(task)
+
+        private fun temporarilyIgnoreAsWorkaround(task: TaskInternal): Boolean {
```
Comment (bamboo) [inline subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/TaskExecutionAccessCheckers.kt:65]: 🔧 Move it to `Workarounds`.

**Comment by h0tk3y** [comment]
@bot-gradle test RFN

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractAndroidSantaTrackerSmokeTest.groovy [hunk 2]
```diff
@@ -74,7 +74,9 @@ class AbstractAndroidSantaTrackerSmokeTest extends AbstractSmokeTest {
     protected SmokeTestGradleRunner runnerForLocation(File projectDir, String agpVersion, String... tasks) {
         def runnerArgs = [[
             // TODO: the versions of KGP we use still access Task.project from a cacheIf predicate
-            "-Dorg.gradle.configuration-cache.internal.task-execution-access-pre-stable=true",
+            // A workaround for this has been added to TaskExecutionAccessCheckers;
```
Comment (bamboo) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractAndroidSantaTrackerSmokeTest.groovy:77]: 👍

**Comment by h0tk3y** [comment]
@bot-gradle test RFN

File: subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/TaskExecutionAccessCheckers.kt (hunk 1 above)
Comment (h0tk3y) [reply subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/TaskExecutionAccessCheckers.kt]: Done in https://github.com/gradle/gradle/compare/d23ff36fd2173c72b4ecadbb5030cc23418a10c1..d478204088a6b293046eba396f91f7c2bb3cad6d

**Review by bamboo — APPROVED** [review]

**Comment by h0tk3y** [comment]
@bot-gradle test and merge

**Comment by bamboo** [comment]
@bot-gradle test and merge

**Pull request merged**